crates/polars-core/src/chunked_array/comparison/categorical.rs

```diff
@@ -30,20 +30,68 @@ where
     Ok(compare_function(lhs.physical(), rhs))
 }
 
-fn cat_compare_helper<'a, Compare>(
+fn cat_compare_helper<'a, Compare, CompareString>(
     lhs: &'a CategoricalChunked,
     rhs: &'a CategoricalChunked,
     compare_function: Compare,
+    compare_str_function: CompareString,
 ) -> PolarsResult<BooleanChunked>
 where
     Compare: Fn(&'a UInt32Chunked, &'a UInt32Chunked) -> BooleanChunked,
+    CompareString: Fn(&str, &str) -> bool,
 {
     let rev_map_l = lhs.get_rev_map();
     let rev_map_r = rhs.get_rev_map();
-    polars_ensure!(rev_map_l.is_enum() && rev_map_r.is_enum(), ComputeError: "can not compare (<, <=, >, >=) two categoricals, unless they are of Enum type");
-    polars_ensure!(rev_map_l.same_src(rev_map_r), ComputeError: "can only compare Enum types with the same categories {:?} vs {:?}", rev_map_l.get_categories(), rev_map_r.get_categories());
+    polars_ensure!(rev_map_l.same_src(rev_map_r), ComputeError: "can only compare categoricals of the same type and the same categories {:?} vs {:?}", rev_map_l.get_categories(), rev_map_r.get_categories());
```

Comment:
There might be lots of categories, should we maybe refrain from printing the categories themselves?

Reply:
Yeap, good point.